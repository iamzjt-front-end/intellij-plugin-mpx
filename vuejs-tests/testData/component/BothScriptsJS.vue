<script>
export class Struct {
  a = 1;
  b = "hello";
}

export default {
  name: "Rename",
};
</script>

<script setup>
defineProps({
  optional: Struct,
});

<weak_warning descr="Unresolved variable or type optional">optional</weak_warning>;
</script>

<template>
  {{optional}}
  <!-- Unresolved warning is the WebStorm bug -->
  <Rename
    :optional="new <weak_warning descr="Unresolved type Struct">Struct</weak_warning>()"
    v-if="optional === undefined"
  />
</template>
