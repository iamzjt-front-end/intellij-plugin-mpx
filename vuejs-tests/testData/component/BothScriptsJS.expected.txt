{
 class: "VueSourceComponent",
 defaultName: "Rename",
 components: {
  BothScriptsJS: <recursive value of class VueSourceComponent>,
 },
 props: [
  {
   class: "VueSourceInputProperty",
   name: "optional",
   jsType: "Struct",
   required: false,
  },
 ],
}
